src/benchmark_mediation.py

```diff
@@ -617,174 +619,309 @@ def alternative_estimator(y, t, m, x, regularization=True):
 
-    t       array-like, shape (n_samples)
-            treatment value for each unit, binary
+    m : array-like, shape (n_samples)
+        Mediator value for each unit, necessarly binary and unidimensional
 
-    m       array-like, shape (n_samples)
-            mediator value for each unit, here m is necessary binary and uni-
-            dimensional
+    x : array-like, shape (n_samples, n_features_covariates)
+        Covariates value for each unit, continuous
 
-    x       array-like, shape (n_samples, n_features_covariates)
-            covariates (potential confounders) values
+    interaction : boolean, default=False
+        Whether to include interaction terms in the model
+        interactions are terms XT, TM, MX
 
-    interaction boolean, default=False
-                whether to include interaction terms in the model
-                interactions are terms XT, TM, MX
+    forest : boolean, default=False
+        Whether to use a random forest model to estimate the propensity
+        scores instead of logistic regression, and outcome model instead
+        of linear regression
 
-    penalty string {‘l1’, ‘l2’, ‘elasticnet’, ‘none’}, default=’l2’
-            passed to the scikit learn sklearn.linear_model.LogisticRegression
-            model. Was used to mimick the R implementation without penalty in
-            some cases, to evaluate regularization bias
+    crossfit : integer, default+0
+        Number of folds for cross-fitting. If crossfit<2, no cross-fitting is applied
 
-    forest  boolean, default False
-            whether to use a random forest model to estimate the propensity
-            scores instead of logistic regression, and outcome model instead
-            of linear regression
+    clip : float, default=0.01
+        Limit to clip p_x and f_mtx for numerical stability (min=clip, max=1-clip)
 
-    crossfit integer, default 0
-             number of folds for cross-fitting
+    regularization : boolean, default=True
+        Whether to use regularized models (logistic or linear regression).
+        If True, cross-validation is used to chose among 8 potential
+        log-spaced values between 1e-5 and 1e5
 
-    clip    float
-            limit to clip for numerical stability (min=clip, max=1-clip)
+    calibration : boolean, default=True
+        [...]
+
+    calib_method : str, default="sigmoid"
+        [...]
 
-    regularization boolean, default True
-                   whether to use regularized models (logistic or
-                   linear regression). If True, cross-validation is used
-                   to chose among 8 potential log-spaced values between
-                   1e-5 and 1e5
 
+    Returns
+    -------
+    list
+        A list of estimated effects :
+        [total effect,
+        direct effect on the exposed,
+        direct effect on the unexposed,
+        indirect effect on the exposed,
+        indirect effect on the unexposed,
+        number of clipped samples]
```

Comment:
not sure this is the wanted behavior, the last returned value should have the same meaning in all methods, so the number of discarded observations by trimming, not an alternance of clipped and trimmed examples. WDYT @sami6mz @bthirion?

Reply:
+1 for consistency

Reply:
More importantly, there should be 6 output arguments, not one list.

Reply:
> not sure this is the wanted behavior, the last returned value should have the same meaning in all methods, so the number of discarded observations by trimming, not an alternance of clipped and trimmed examples. WDYT @sami6mz @bthirion?

Then should we keep clipping? Or replace clipping by trimming?

Reply:
I ended up removing clipping count with [df600a9](https://github.com/judithabk6/med_bench/pull/35/commits/df600a903fce6c8bb932bf2468326e92d939a5b7) + putting the clipping code in issue #39

Reply:
> keep clipping but not return the number of clipped examples. And potentially open an issue to implement trimming in this method. And another issue to implement clipping in all methods and return 6 results instead of 5 in that case

Let's just implement trimming for now, and remove clipping